plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -847,9 +851,11 @@ public Optional<ConnectorOutputMetadata> finishInsert(ConnectorSession session,
 
         IcebergWritableTableHandle table = (IcebergWritableTableHandle) insertHandle;
         Table icebergTable = transaction.table();
+        Optional<Long> previousSnapshotId = Optional.ofNullable(icebergTable.currentSnapshot()).map(Snapshot::snapshotId);
+        Schema schema = icebergTable.schema();
```

Comment:
Extracting schema to a variable looks like it should belong to a preparatory commit